src/pages/LinkCustomer.tsx

```diff
@@ -0,0 +1,73 @@
+import { FC, ChangeEvent, useState } from "react";
+import { useDebouncedCallback } from "use-debounce";
+import { useDeskproAppClient, Button } from "@deskpro/app-sdk";
+import { useStore } from "../context/StoreProvider/hooks";
+import { setEntityCustomer } from "../services/entityAssociation";
+import { getCustomers } from "../services/shopify";
+import { Customer } from "../components/Customer";
+
+const NoResult = () => (
+    <p>No matching customers found. Please try again.</p>
+);
+
+export const LinkCustomer: FC = () => {
+    const [state, dispatch] = useStore();
+    const { client } = useDeskproAppClient();
+    const [searchQuery, setSearchQuery] = useState<string>("");
+    const [customers, setCustomers] = useState<[]>([]);
+    const [selectedCustomerId, setSelectedCustomerId] = useState<string>('');
+    const primaryUser = state.context?.data.ticket?.primaryUser;
+
+    const searchInShopify = useDebouncedCallback<(q: string) => void>((q) => {
+        if (!client) {
+            return;
+        }
+
+        if (!q || q.length < 2) {
+            setCustomers([]);
+            return;
+        }
+
+        getCustomers(client, { querySearch: q })
+            .then(({ customers }: { customers: [] }) => {
+                if (Array.isArray(customers)) {
+                    setCustomers(customers);
+                }
+            })
+            .catch(() => {});
+    },500);
+
+    const onChangeSearch = ({ target: { value: q } }: ChangeEvent<HTMLInputElement>) => {
+        setSearchQuery(q);
+        searchInShopify(q);
+    };
+
+    const onChangeSelectedCustomer = (e: ChangeEvent<HTMLInputElement>) => {
+        setSelectedCustomerId(e.target.value);
+    }
+
+    const onAdd = () => {
+        if (!client) {
+            return;
+        }
+
+        setEntityCustomer(client, primaryUser.id, selectedCustomerId)
+            .then(() => dispatch({ type: "changePage", page: "home" }))
+            // ToDo: handle error
+            .catch(() => {});
```

Comment:
For error handling we need to do two things: 

1. Alert the user that something has failed, and give them some actions; to either ignore it and carry on, or to inform them that the error is critical and that they should reload the app
2. Log out a more verbose error to the console via `console.error()` so we can debug

When alerting to users, in the JIRA app we set this in the global store and render to the user on any page: https://github.com/DeskproApps/jira/blob/master/src/context/StoreProvider/types.ts#L32

Reply:
done